src/mdsynthesis/treants.py

```diff
@@ -71,57 +71,31 @@ def universe(self):
         the universe definition entirely.
 
         """
-        # TODO: include check for changes to universe definition, not just
-        # definition absence
-        if self._universe:
-            return self._universe
-        else:
-            self.universedef._activate()
-            return self._universe
+        _args = self.universedef._args
```

Comment:
I'm using the args as a hash now. This ensures the universe is updated when topology or trajectory change.

Reply:
I don't think we want this in the `__init__`  method, since it means we will be loading a Universe on init of the `Sim` if it's defined. The point of the `Sim.universe` property is that this loads the universe on access if it's not already loaded, since for big universes or those with lots of trajectories that need to be indexed, this can be very slow.

Reply:
this is not in the init method. But we can reintroduce the lazy loading. I didn't think about that when I made the change.

Reply:
@kain88-de because I just removed it. Got lazy loading back so we're good on this front. Working on discover mechanics now.